baseplate/integration/pyramid.py

```diff
@@ -118,17 +111,13 @@ def _on_new_request(self, event):
                     flags=flags,
                 )
 
-                authn_token = request.headers.get("X-Authentication", None)
-                request_context_payload = request.headers.get("X-Edge-Request",
-                                                              None)
-                authentication_context = \
-                    self.auth_factory.make_context(authn_token)
-                authentication_context.attach_context(request)
-                request_context = EdgeRequestContext(
-                    header=request_context_payload,
-                    authentication_context=authentication_context,
-                )
-                request_context.attach_context(request)
+                edge_payload = request.headers.get("X-Edge-Request", None)
+                if self.edge_context_factory:
+                    edge_context = self.edge_context_factory.from_upstream(
+                        edge_payload)
+                    edge_context.attach_context(request)
+                else:
+                    request.raw_request_context = edge_payload
```

Comment:
what's happening here?

Reply:
added a comment to try and explain, does that help? would it be better to structure this differently so i don't need the same branching in both integrations? or not worth the added abstraction?

Reply:
ok I get it. I think it's fine